DTS-STN/canadian-dental-care-plan
# AB#2659 -- Add authentication check to every page

## Apologies for the size of this PR 😞

### Description

This PR adds an authenticate check to every page in the application. An authentication check involves:

1. Checking to see if we have a CDCP session.
1. Checking to see if we have an RAOIDC session.

If either of these checks fail, the application will forward the user to RAOIDC to complete the login flow.

### Related Azure Boards Work Items

- AB#2673
- AB#2659
- AB#2916
- AB#2670
- AB#2742
- AB#2918

### Checklist


### Test Instructions

It is tricky to test this, since most of the RAOIDC stuff is mocked. One way would be to open your browser's devtools before hitting the application, then visit <http://localhost:3000/> and watch the requests in the network tab. You will see a mocked RAOIDC login happen.
